### DISCUSSION TITLE
Will ukv ever have support for queues?

### DISCUSSION THREAD

**Discussion by kesavkolla** [unum-cloud/UStore#228] [Q&A] [answered]
I am looking ukv as an alternative for redis. Today redis support queues also. ukv in general great for addressing different modalities. It would be good to add queues also.

**Comment by ashvardanian** [comment, marked as answer]
Potentially, yes. However, series and queues are dissimilar in their writing patterns. The first is append-only, and the latter is first-in-first-out. The best structure for a queue is a ring buffer, not a key-value store. It can be added for compatibility, but it may not be the best idea right away.

Let's keep this conversation open and see if there is more demand for this to be prioritized.

**Comment by kesavkolla** [comment]
Thanks for the reply. I hope at some point ukv will be a contender to replace redis. Today's key/value market share is dominent by redis. Some of other projects like kvrocks, tidis offer redis compatibility. It would be nice to have an ability to add redis as a compatibility layer. This way we can use existing redis libraries for and without coding chnages we can start using ukv.

**Reply by ashvardanian** [reply]
I think a Redis and Postgres compatibility can be implemented as a separate repo, using UKV under the hood. That would be a better path to compatibility, than fitting everything here.
